game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,30 +2,63 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root";
+    private static final int MAX_DEPTH_GUESS = 4;
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        try {
+            addAllAssets(getResourceRoot());
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("No resource root found.");
+            System.exit(0);
+        }
     }
 
+    private FileHandle getResourceRoot() throws IOException {
+        Predicate<Path> filter =
+                p ->
+                        Files.isRegularFile(p)
+                                && PLACEHOLDER_FILENAME.equals(p.getFileName().toString());
+        try (Stream<Path> walk =
```

Comment:
dieses try, wozu ist das gut?

Reply:
Files.walk ist ein Auto-Closeable... Diese sollten immer im try-with-resource stehen... Siehe hier https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#obl-method-may-fail-to-clean-up-stream-or-resource-obl-unsatisfied-obligation

Ich fände es ohne try... eigentlich auch besser